### DISCUSSION TITLE
Sharding A Minestom Server

### DISCUSSION THREAD

**Discussion by DersWasTaken** [Minestom/Minestom#1700] [Help]
Hey! First time back in minestom since I finally got some time after a couple months.

Wanting to start off with some basic implementation of things, so im wondering if theres a way to 'shard' a minestom server; for example I would launch a main server, which say has 4GB/32GB ram, and then somehow* create another server which would also have 4GB but be completely separate from the other server. This is similar to hypixels instance system, but I'd like to create them dynamically as needed based on how many servers I need to allocate towards a certain instance(s). If you need any further information, dont hesitate to ask.

Thanks all!

**Comment by emortaldev** [comment]
Sounds like you want Kubernetes or something closely related to Docker since you said completely separate. Hypixel also likely uses Kubernetes so should be similar to what you are looking at.

**Comment by dev-hydrogen** [comment]
Might i suggest you try using [Minestom Instances](https://wiki.minestom.net/world/instances)? You could have a "server instance" (which would be managed with kubernetes, as emortal suggested) with for example 4 minestom instances on it (4 entirely different worlds) hosting whatever you want
This solution might be harder to program, but also may be more performant (theres no server overhead for every single instance)

This project might be of interest: https://github.com/FlexEGames/FlexEGames, though i have not explored it more than looking at the github page.

**Reply by dev-hydrogen** [reply]
P.S 4gb ram dedicated for a minestom server is a lot, unless you plan on having like 50-200 players on that server, or your extensions take up a lot of ram for whatever reason

**Reply by zax71** [reply]
500mb is a lot more normal for a Minestom server
